compiler/extensions/cpp/runtime/test/zserio/BitStreamReaderTest.cpp

```diff
@@ -26,6 +26,32 @@ class BitStreamReaderTest : public ::testing::Test
 
 const size_t BitStreamReaderTest::BUFFER_SIZE;
 
+TEST_F(BitStreamReaderTest, bitBufferCtor)
+{
+    uint8_t data[] = {1, 2, 3};
+
+    zserio::BitBuffer buffer(data, 17u);
+    zserio::BitStreamReader reader(buffer);
+
+    ASSERT_EQ(17u, reader.getBufferBitSize());
+}
+
+TEST_F(BitStreamReaderTest, readUnalignedData)
+{
+    /* 1bit = 0, 8bit = 8, 8bit = 1 */
+    uint8_t data[] = {0x04, 0x00, 0x80};
+
+    zserio::BitBuffer buffer(data, 17u);
+    zserio::BitStreamReader reader(buffer);
+
+    auto readBool = reader.readBool();
+    ASSERT_EQ(false, readBool);
+
+    zserio::BitBuffer readBuffer = reader.readBitBuffer();
+    ASSERT_EQ(8u, readBuffer.getBitSize());
+    ASSERT_EQ(1u, readBuffer.getBuffer()[0]);
+}
```

Comment:
This seems to be only related BitBuffer which has currently a strange logic that it uses the last byte bits from the LSB. E.g. when you have a buffer containing 10 bits, it use the full first byte, but only 2 lowest bits from the second byte. It would be better to use the two highest bits since it's more intuitive and even more consistent. I don't understand yet how the fix in the BitStreamReader::loadCacheNext will affect it. Since we will probably change it, we will check this PR once working on it.

Reply:
I've changed the test to not use `BitBuffer` but `readBits` instead – same behaviour.